### PR TITLE
Gi gui

### PR DESCRIPTION
**Pull request by kroczi**

### CODE REVIEW

File: gi_gui/MainFrame.cpp [hunk 1]
```diff
@@ -0,0 +1,576 @@
...
+  EVT_TEXT(ID_RefA, MainFrame::OnRefA)
+  EVT_TEXT(ID_RefB, MainFrame::OnRefB)
+  EVT_TEXT(ID_TarA, MainFrame::OnTarA)
+  EVT_TEXT(ID_TarB, MainFrame::OnTarB)
+wxEND_EVENT_TABLE()
+
+void MainFrame::OnReference(wxCommandEvent& event)
+{
+  wxFileDialog openFileDialog(this, _("Open reference file"), "", "",
+      "DICOM files (*.dcm)|*.dcm", wxFD_OPEN|wxFD_FILE_MUST_EXIST);
```
Comment (kroczi) [inline gi_gui/MainFrame.cpp]: add also *.IMA extention

File: gi_gui/MainFrame.cpp [hunk 2]
```diff
@@ -0,0 +1,576 @@
...
+    return;
+
+  wxFileOutputStream output_stream(saveFileDialog.GetPath());
+  if(!output_stream.IsOk())
+  {
+    wxLogError("Cannot save current contents in file '%s'.", saveFileDialog.GetPath());
+    return;
+  }
+
+  if(lastComparison == NULL)
```
Comment (kroczi) [inline gi_gui/MainFrame.cpp]: maybe instead of this, the save button should be disabled before first comparison.

File: gi_gui/MainFrame.cpp [hunk 3]
```diff
@@ -0,0 +1,576 @@
...
+  if(saveModes->GetValue().ToStdString().compare("All except private") == 0)
+  {
+    strategy = 3;
+  }
+
+  set <pair<int, int>> emptyTagSet = {};
+
+  saveImage(lastComparison->dims, lastComparison->refDataSet, saveFileDialog.GetPath().ToStdString(), lastComparison->gamma, lastComparison->refXNumber, lastComparison->refYNumber, lastComparison->refZNumber, lastComparison->rescale, lastComparison->info, strategy, emptyTagSet);
+
+  giPath = saveFileDialog.GetPath();
```
Comment (kroczi) [inline gi_gui/MainFrame.cpp]: what is this variable for?

File: gi_gui/MainFrame.cpp [hunk 4]
```diff
@@ -0,0 +1,576 @@
...
+  setMin(std::to_string(imageMin(size, gamma)));
+  setAvg(std::to_string(imageAvg(size, gamma)));
+  setMax(std::to_string(imageMax(size, gamma)));
+  setGPR(std::to_string(gammaPassingRate(size, gamma)));
+
+  lastComparison = new Comparison(parsedParameters->dims, CodecFactory::load(refPath.ToStdString()), gamma,
+    refXNumber, refYNumber, refZNumber, parsedParameters->rescale, info);
+}
+
+int MainFrame::getNumberOfPoints(int refXNumber, int refYNumber, int refZNumber)
```
Comment (kroczi) [inline gi_gui/MainFrame.cpp]: there is already appropriate function for this in additions.

File: gi_gui/MainFrame.cpp [hunk 5]
```diff
@@ -0,0 +1,576 @@
...
+}
+
+void MainFrame::OnStatusLabel(wxCommandEvent& event)
+{
+  status->Refresh();
+  string state = status->GetValue().ToStdString();
+
+  if(state.compare("Calculating") == 0)
+  {
+
```
Comment (kroczi) [inline gi_gui/MainFrame.cpp]: ??

File: gi_gui/MainFrame.cpp [hunk 6]
```diff
@@ -0,0 +1,576 @@
...
+{
+  setState("Waiting");
+}
+
+void MainFrame::OnFilter(wxCommandEvent& event)
+{
+  setState("Waiting");
+}
+
+bool MainFrame::checkForConfigCorrectioness()
```
Comment (kroczi) [inline gi_gui/MainFrame.cpp]: already discussed this

File: gi_gui/MainFrame.cpp [hunk 7]
```diff
@@ -0,0 +1,576 @@
...
+  what->AppendText(wxString(value));
+}
+
+void MainFrame::refreshFileWithConfigs()
+{
+  string confsPath = "./config";
+
+  std::ofstream outfile (confsPath);
+
+  for (size_t i = 0; i < configItems.GetCount(); i++)
```
Comment (kroczi) [inline gi_gui/MainFrame.cpp]: how does it work? where is configItems filled?

File: gi_gui/ConfigDialog.cpp [hunk 8]
```diff
@@ -0,0 +1,277 @@
+#include "ConfigDialog.h"
+#include "MainFrame.h"
+
+
+ConfigDialog::ConfigDialog(MainFrame* parent, const wxString& title)
+  : wxDialog(NULL, -1, title)
+{
+  parent_ = parent;
```
Comment (kroczi) [inline gi_gui/ConfigDialog.cpp]: to initialization list

File: gi_gui/ConfigDialog.cpp [hunk 9]
```diff
@@ -0,0 +1,277 @@
...
+  }
+
+  string path = "./configs/" + pick.ToStdString() + ".config";
+
+  ifstream inputFile;
+  inputFile.open(path);
+
+  string boofor;
+
+  if(inputFile.is_open())
```
Comment (kroczi) [inline gi_gui/ConfigDialog.cpp]: maybe, extract this  to separate function

File: gi_gui/Comparison.cpp [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+  int refXNumber_, int refYNumber_, int refZNumber_, int rescale_, string info_)
+{
+    dims = dims_;
+    refDataSet = refDataSet_;
+    gamma = gamma_;
+    refXNumber = refXNumber_;
+    refYNumber = refYNumber_;
+    refZNumber = refZNumber_;
+    rescale = rescale_;
+    info = info_;
```
Comment (kroczi) [inline gi_gui/Comparison.cpp]: initialization list

File: gi_gui/MainFrame.cpp (hunk 3 above)
Comment (wojciech-zagrajczuk) [inline gi_gui/MainFrame.cpp]: It was needed for debugging this save GI problem. Problably it's not needed anymore. Gonna find out and fix it if needed

File: gi_gui/makefile [hunk 11]
```diff
@@ -0,0 +1,31 @@
+# IMEBRA
+IMEBRA_INC_PATH=/home/afetrist/libs/imebra/library/include
```
Comment (kroczi) [inline gi_gui/makefile]: change to relative path

File: gi_gui/MainFrame.cpp [hunk 12]
```diff
@@ -0,0 +1,587 @@
...
+void MainFrame::OnCalculate(wxCommandEvent& event)
+{
+  setState("Calculating");
+  Parameters* parsedParameters = parser->resolveParameters(dims->GetValue(), dd->GetValue(), dta->GetValue(), refval->GetValue(),
+    local->GetValue(), limit->GetValue(), rescale->GetValue(), plane->GetValue(), refSlice->GetValue(), tarSlice->GetValue(),
+    refA->GetValue(), refB->GetValue(), tarA->GetValue(), tarB->GetValue());
+  int refNDims, tarNDims, refXNumber, refYNumber, refZNumber, tarXNumber, tarYNumber, tarZNumber;
+  double refXStart, refYStart, refZStart, tarXStart, tarYStart, tarZStart,
+          refXSpacing, refYSpacing, refZSpacing, tarXSpacing, tarYSpacing, tarZSpacing;
+  unique_ptr<imebra::DataSet> refDataSet(CodecFactory::load(refPath.ToStdString()));
```
Comment (kroczi) [inline gi_gui/MainFrame.cpp]: loadDicom

File: gi_gui/MainFrame.cpp [hunk 13]
```diff
@@ -0,0 +1,587 @@
...
+  unique_ptr<imebra::DataSet> tarDataSet(CodecFactory::load(tarPath.ToStdString()));
+  double* reference = acquireImage(refDataSet.get(), refNDims,
+                                   refXStart, refXSpacing, refXNumber,
+                                   refYStart, refYSpacing, refYNumber,
+                                   refZStart, refZSpacing, refZNumber);
+  double* target = acquireImage(tarDataSet.get(), tarNDims,
+                                tarXStart, tarXSpacing, tarXNumber,
+                                tarYStart, tarYSpacing, tarYNumber,
+                                tarZStart, tarZSpacing, tarZNumber);
+  if (refNDims != tarNDims || (tarNDims != parsedParameters->dims && (tarNDims != 3 || parsedParameters->dims != 2.5)))
```
Comment (kroczi) [inline gi_gui/MainFrame.cpp]: dims should be ints

File: gi_gui/MainFrame.cpp [hunk 14]
```diff
@@ -0,0 +1,587 @@
...
+                                 refYStart, refYSpacing, refYNumber,
+                                 refZStart, refZSpacing, refZNumber,
+                                 tarXStart, tarXSpacing, tarXNumber,
+                                 tarYStart, tarYSpacing, tarYNumber,
+                                 tarZStart, tarZSpacing, tarZNumber,
+                                 parsedParameters->dd, parsedParameters->dta, parsedParameters->local,
+                                 parsedParameters->refVal, parsedParameters->limit);
+  setState("Finished");
+
+  string info =  createComputationDescription(refPath.ToStdString(), tarPath.ToStdString(), 2);
```
Comment (kroczi) [inline gi_gui/MainFrame.cpp]: 2 -> algorithm

File: gi_gui/MainFrame.cpp [hunk 15]
```diff
@@ -0,0 +1,587 @@
...
+  double* target = acquireImage(tarDataSet.get(), tarNDims,
+                                tarXStart, tarXSpacing, tarXNumber,
+                                tarYStart, tarYSpacing, tarYNumber,
+                                tarZStart, tarZSpacing, tarZNumber);
+  if (refNDims != tarNDims || (tarNDims != parsedParameters->dims && (tarNDims != 3 || parsedParameters->dims != 2.5)))
+  {
+      cout << "Dims mismatch!" << endl;
+      exit(-1);
+  }
+  int algVer = 2;
```
Comment (kroczi) [inline gi_gui/MainFrame.cpp]: algVer -> algorithm

File: gi_gui/makefile [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+# GAMMA ADDITIONS
+GAMMA_ADDITIONS_INC_PATH=../gi_additions
+GAMMA_ADDITIONS_LINK_PATH=../gi_additions/bin
+GAMMA_ADDITIONS_LIB=gamma_additions
+
+CORE_INC_LINK_FLAGS = -I$(GAMMA_INC_PATH) -L$(GAMMA_LINK_PATH) -l$(GAMMA_LIB)
+
+GAMMA_ADDITIONS_INC_LINK_FLAGS = -I$(GAMMA_ADDITIONS_INC_PATH) -L$(GAMMA_ADDITIONS_LINK_PATH) -l$(GAMMA_ADDITIONS_LIB)
+
+WRAPPER_INC_LINK_FLAGS = -I$(IMEBRA_INC_PATH) -L$(IMEBRA_LINK_PATH) -l$(IMEBRA_LIB) -I$(GAMMA_INC_WRAPPER_PATH) -L$(GAMMA_LINK_WRAPPER_PATH) -l$(GAMMA_WRAPPER_LIB) $(CORE_INC_LINK_FLAGS) $(GAMMA_ADDITIONS_INC_LINK_FLAGS)
```
Comment (kroczi) [inline gi_gui/makefile]: move imebra flags to the end
othervise loadDicom won't work

**Pull request merged**